### PR TITLE
Upgrade specs2 to 3.6

### PR DESCRIPTION
**Pull request by 2rs2ts**
A nice retry of #116 :)

* Upgraded specs2
* In order to make versions not conflict, scalacheck upgraded to 1.12.2 and spray changed to the version which uses shapeless2 (https://github.com/spray/spray/issues/1008)
* Tests using Futures and `await` migrated to use an [`ExecutionEnvironment`](https://etorreborre.github.io/specs2/guide/SPECS2-3.6/org.specs2.guide.ExecutionEnvironments.html)
* `ActorSpecification` moved to using the `ActionDsl` in order to allow the custom `Step` which shuts down the actor system at the end of the test
* Removes our custom `NoConcurrentExecutionContext` because the underlying `ConcurrentExecutionContext` was deprecated.

### CODE REVIEW

**Comment by kionka** [comment]
I love version bumps! :shipit: after the build passes.

File: common/src/test/scala/com/paypal/cascade/common/tests/option/RichOptionSpecs.scala [hunk 1]
```diff
@@ -55,7 +57,7 @@ class RichOptionSpecs extends Specification { override def is = s2"""
     }
   }
 
-  case class ToFuture() extends Context {
+  case class ToFuture(implicit ev: ExecutionEnv) extends Context {
```
Comment (kionka) [inline common/src/test/scala/com/paypal/cascade/common/tests/option/RichOptionSpecs.scala]: What's going on with these ```implicit ev: ExecutionEnv``` parameters?

File: common/src/test/scala/com/paypal/cascade/common/tests/util/NoConcurrentExecutionContext.scala [hunk 2]
```diff
@@ -1,27 +0,0 @@
...
-
-import org.specs2.matcher.ConcurrentExecutionContext
-import scala.concurrent.ExecutionContext
-
-/**
- * This trait can be used to deactive the implicit concurrent execution context.
- * Added in Specs 2.4 via https://github.com/etorreborre/specs2/commit/a165db45ba213f1794b3738ab996aad383a1ba0e.
- */
-trait NoConcurrentExecutionContext extends ConcurrentExecutionContext {
-  override val concurrentExecutionContext: ExecutionContext = concurrent.ExecutionContext.Implicits.global
```
Comment (2rs2ts) [inline common/src/test/scala/com/paypal/cascade/common/tests/util/NoConcurrentExecutionContext.scala]: The reason this was removed was due to this deprecation: https://github.com/etorreborre/specs2/blob/specs2-three/matcher/src/main/scala/org/specs2/matcher/ConcurrentExecutionContext.scala#L9

File: common/src/test/scala/com/paypal/cascade/common/tests/option/RichOptionSpecs.scala (hunk 1 above)
Comment (2rs2ts) [inline common/src/test/scala/com/paypal/cascade/common/tests/option/RichOptionSpecs.scala]: https://etorreborre.github.io/specs2/guide/SPECS2-3.6/org.specs2.guide.ExecutionEnvironments.html

**Comment by kionka** [comment]
So many implicits. :disappointed:

**Pull request merged**